smartpy/losses/classification_losses.py

```diff
@@ -0,0 +1,16 @@
+from smartpy.interfaces.loss import Loss
+
+import theano.tensor as T
+
+
+class NegativeLogLikelihood(Loss):
+    def _loss_function(self, model_output):
+        nll = -T.log(model_output)
+        indices = T.cast(self.target[:, 0], dtype="int32")  # Targets are floats.
+        selected_nll = nll[T.arange(self.target.shape[0]), indices]
```

Comment:
Not sure I understand what you are saying.
`self.target` has a shape of `(nb_examples, 1) and `nll`is matrix of shape `(nb_examples, nb_classes)`, so we have to access the matrix `nll` n=nb_examples times, once per row.